tasks/gulp/asset-version.js

```diff
@@ -1,35 +1,49 @@
 'use strict'
 
 const configPaths = require('../../config/paths.json')
-const gulp = require('gulp')
 const fs = require('fs')
+const path = require('path')
 const taskArguments = require('./task-arguments')
-const gulpif = require('gulp-if')
-const rename = require('gulp-rename')
-const del = require('del')
-const vinylPaths = require('vinyl-paths')
 
 // check for the flag passed by the task
-
 const isDist = taskArguments.destination === 'dist' || false
 
 // Update assets' versions ----------
 // Add all.package.json version
 // ----------------------------------
-gulp.task('update-assets-version', () => {
+function updateAssetsVersion (cb) {
   const pkg = require('../../' + configPaths.package + 'package.json')
+
+  // Write VERSION.txt file
   fs.writeFileSync(taskArguments.destination + '/VERSION.txt', pkg.version + '\r\n')
-  return gulp.src([
+
+  // Files to process
+  const assetFiles = [
     taskArguments.destination + '/govuk-frontend.min.css',
     taskArguments.destination + '/govuk-frontend-ie8.min.css',
     taskArguments.destination + '/govuk-frontend.min.js'
-  ])
-    .pipe(vinylPaths(del))
-    .pipe(gulpif(isDist,
-      rename(obj => {
-        obj.basename = obj.basename.replace(/(govuk.*)(?=\.min)/g, '$1-' + pkg.version)
-        return obj
-      })
-    ))
-    .pipe(gulp.dest(taskArguments.destination + '/'))
-})
+  ]
+
+  // Loop through files and move them to their new locations
+  assetFiles.map(srcFilename => {
+    let destFilename = srcFilename
+    if (isDist) {
```

Comment:
Does this function ever run outside of `dist`?

Wondering if we should get really specific and call this something like `updateDistAssetsVersion` and get rid of this `if` clause. We could probably then drop the need to copy the files to the same location.

Reply:
It doesn't *seem* like this task is used for anything else in our own build:dist task, and that would seem to be supported by it not really doing anything but write `VERSION.txt` if the if-statement fails.

Changing the destination folder is an option we provide through the command line though, so I'm not 100% confident that the option is never used.

Reply:
Yeah, as far as I can see, we just use that command line option to separate the package and dist builds.

I think we should simplify our tasks and get rid of the `destination` option, but maybe that's a step for after we've updated most of these.

From what I can see:

- The only task in common for `build:package` and `build:dist` is `clean`, and that does different things depending on which destination is set, so it probably makes sense just to make them two separate tasks.
- The `copy-to-destination` file uses `destination`, but its `copy-files` task is only used in `build:package`
- The `compile-assets` file uses `destination`, but again, that whole file is a bit of a mess and we could simplify it.
- `copy:assets` is, supposedly, supposed to be used to copy to `public`, but it's actually only used for the `dist` build task
- The only times we run a destination-reliant thing using the default of `public` are in the `heroku` and `test` scripts, so it feels like we'd probably be able to simplify that as well.

Reply:
I've removed the destination if-statement. 👍